crates/influxive-writer/tests/file.rs

```diff
@@ -0,0 +1,138 @@
+use crate::common::telegraf_influx_file_conf::TelegrafLineProtocolConfigBuilder;
+use crate::common::telegraf_svc::TelegrafSvc;
+use influxive_child_svc::{InfluxiveChildSvc, InfluxiveChildSvcConfig};
+use influxive_core::*;
+use influxive_writer::*;
+use std::path::{Path, PathBuf};
+use std::time::Duration;
+
+mod common;
+
+/// Setup InfluxiveWriter to use LineProtocolFileBackendFactory
+pub fn create_influx_file_writer(test_path: &PathBuf) -> InfluxiveWriter {
+    let _ = std::fs::remove_file(&test_path);
+    let mut config =
+        InfluxiveWriterConfig::create_with_influx_file(test_path.clone());
+    config.batch_duration = std::time::Duration::from_millis(30);
+    let writer = InfluxiveWriter::with_token_auth(config.clone(), "", "", "");
+    writer
+}
+
+/// Spawn influxDB with the default config
+async fn spawn_influx(path: &Path) -> InfluxiveChildSvc {
+    let child = InfluxiveChildSvc::new(
+        InfluxiveChildSvcConfig::default()
+            .with_database_path(Some(path.into()))
+            .with_metric_write(
+                InfluxiveWriterConfig::default()
+                    .with_batch_duration(std::time::Duration::from_millis(5)),
+            ),
+    )
+    .await
+    .unwrap();
+
+    child.ping().await.unwrap();
+
+    child
+}
+
+async fn write_metrics_to_file(test_path: &PathBuf) {
```

Comment:
I find this function with the assertion odd. I prefer simpler tests where you test one thing.
In this case, I suggest you write one test which checks that the writer can write metrics to a file. Possibly even assert the exact line(s) in the file that you expect there to be.

Then another test can write metrics to disk/file and it's tested that the influx reader can query them. That one you already have and I think it's good, just that this helper function is really a separate test for me.

Reply:
I see there's that kind of test already. I meant to extract a shared helper function in case there were two integration tests that use the same test setup. But that's no longer the case.

You could share the test setup code between unit and integration test, for example with a `test_utils` module under `src`. But I'm also fine if the code is duplicated here.